### PR TITLE
Controller -> Credential Types- Add Missing Tests, Refactor to Organize Spec File

### PR DESCRIPTION
**Pull request by akus062381**
This PR does the following:

Restructure the spec file to organize tests
Adds 2 missing tests

### CODE REVIEW

File: cypress/e2e/awx/administration/credentialTypes.cy.ts [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      cy.intercept('PATCH', `api/v2/credential_types/${credType1.id}/`).as('editCredType');
+      cy.clickButton(/^Save credential type$/);
+      cy.wait('@editCredType')
+        .its('response.body.name')
+        .then((name: string) => {
+          expect(editedCredentialTypeName).to.be.equal(name);
+        });
+      cy.verifyPageTitle(editedCredentialTypeName);
+      cy.intercept('DELETE', `api/v2/credential_types/${credType1.id}/`).as('deleteCredType');
+      cy.clickPageAction('delete-credential-type');
```
Comment (nixocio) [inline cypress/e2e/awx/administration/credentialTypes.cy.ts:239]: I could not find the assertion related to the edited items. Do we verify that after was saved the details page show the new name and new description?

File: cypress/e2e/awx/administration/credentialTypes.cy.ts [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      cy.get('[data-cy="name"]').clear().type(editedCredentialTypeName);
+      cy.get('[data-cy="description"]').clear().type('this is a new description after editing');
+      cy.intercept('PATCH', `api/v2/credential_types/${credType1.id}/`).as('editCredType');
+      cy.clickButton(/^Save credential type$/);
+      cy.wait('@editCredType')
+        .its('response.body.name')
+        .then((name: string) => {
+          expect(editedCredentialTypeName).to.be.equal(name);
+        });
+      cy.verifyPageTitle(editedCredentialTypeName);
```
Comment (nixocio) [inline cypress/e2e/awx/administration/credentialTypes.cy.ts:205]: I could not find the assertion related to the edited items. Do we verify that after was saved the details page show the new name and new description?

**Review by nixocio — CHANGES_REQUESTED** [review]
@akus062381 I left a couple of comments. Overall good job.

File: cypress/e2e/awx/administration/credentialTypes.cy.ts (hunk 1 above)
Comment (akus062381) [reply cypress/e2e/awx/administration/credentialTypes.cy.ts:239]: There was an assertion a couple of lines up; I added one additional.

File: cypress/e2e/awx/administration/credentialTypes.cy.ts [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      cy.url().then((currentUrl) => {
+        expect(currentUrl.includes('details')).to.be.true;
+        cy.verifyPageTitle(credType1.name).should('be.visible');
+      });
+    });
+
+    it('can filter credential types by name', () => {
+      cy.navigateTo('awx', 'credential-types');
+      cy.filterTableByMultiSelect('name', [credType1.name]);
+      cy.get('tr').should('have.length.greaterThan', 0);
```
[the remark this answers is not in the document]
Comment (akus062381) [reply cypress/e2e/awx/administration/credentialTypes.cy.ts:62]: Changed

File: cypress/e2e/awx/administration/credentialTypes.cy.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      });
+
+      cy.fixture('credTypes-injector-config').then((credentialType: CredentialType) => {
+        injectorCredType = JSON.stringify(credentialType);
+      });
+    });
+
+    after(() => {
+      cy.deleteAwxCredentialType(credType1, { failOnStatusCode: false });
+      cy.deleteAwxCredential(credential, { failOnStatusCode: false });
```
[the remark this answers is not in the document]
Comment (akus062381) [reply cypress/e2e/awx/administration/credentialTypes.cy.ts:45]: Fixed, thank you

**Review by nixocio — APPROVED** [review]

**Comment by nixocio** [comment]
@akus062381 tests failed on CI after changes.

**Pull request merged**